pyvista/core/utilities/geometric_objects.py

```diff
@@ -383,6 +395,521 @@ def Sphere(
+    radius = np.asanyarray(radius)
+
+    # Hereafter all degrees are in radians
+    # Also radius, phi, theta are now np.ndarrays
+    if theta is None:
+        theta = np.linspace(0, 2 * np.pi, 30)
+    else:
+        if radians:
+            theta = np.asanyarray(theta)
+        else:
+            theta = np.deg2rad(theta)
+
+    if phi is None:
+        phi = np.linspace(0, np.pi, 30)
+    else:
+        if radians:
+            theta = np.asanyarray(theta)
+        else:
+            phi = np.deg2rad(phi)
+
+    nr = len(radius)
+    ntheta = len(theta)
+    nphi = len(phi)
+
+    if nr < 2:
+        raise ValueError("radius resolution must be 2 or more")
+    if ntheta < 2:
+        raise ValueError("theta resolution must be 2 or more")
+    if nphi < 2:
+        raise ValueError("phi resolution must be 2 or more")
+
+    def _is_sorted(a):
+        return np.all(a[:-1] < a[1:])
+
+    if not _is_sorted(radius):
+        raise ValueError("radius is not monotonically increasing")
+    if not _is_sorted(theta):
+        raise ValueError("theta is not monotonically increasing")
+    if not _is_sorted(phi):
+        raise ValueError("phi is not monotonically increasing")
+
+    def _greater_than_equal_or_close(value1, value2):
+        return value1 >= value2 or np.isclose(value1, value2, rtol=0.0, atol=tolerance)
+
+    def _less_than_equal_or_close(value1, value2):
+        return value1 <= value2 or np.isclose(value1, value2, rtol=0.0, atol=tolerance)
+
+    if not _greater_than_equal_or_close(radius[0], 0.0):
+        raise ValueError("minimum radius cannot be negative")
+
+    # range of theta cannot be greater than 360 degrees
+    if not _less_than_equal_or_close(theta[-1] - theta[0], 2 * np.pi):
+        if radians:
+            raise ValueError("max theta and min theta must be within 2 * np.pi")
+        raise ValueError("max theta and min theta must be within 360 degrees")
+
+    if not _greater_than_equal_or_close(phi[0], 0.0):
+        raise ValueError("minimum phi cannot be negative")
+    if not _less_than_equal_or_close(phi[-1], np.pi):
+        if radians:
+            raise ValueError("maximum phi cannot be > np.pi")
+        raise ValueError("maximum phi cannot be > 180.0")
+
+    def _spherical_to_cartesian(r, phi, theta):
+        """Convert spherical coordinate sequences to a ``(n,3)`` Cartesian coordinate array.
+
+        Parameters
+        ----------
+        r : sequence[float]
+            Ordered sequence of floats of radii.
+        phi : sequence[float]
+            Ordered sequence of floats for phi direction.
+        theta : sequence[float]
+            Ordered sequence of floats for theta direction.
+
+        Returns
+        -------
+        np.ndarray
+            ``(n, 3)`` cartesian coordinate array.
+
+        """
+        r, phi, theta = np.meshgrid(r, phi, theta, indexing='ij')
+        x = r * np.sin(phi) * np.cos(theta)
+        y = r * np.sin(phi) * np.sin(theta)
+        z = r * np.cos(phi)
```

Comment:
```suggestion
        x, y, z = pv.spherical_to_cartesian(r, phi, theta)
```

Reply:
Side note: I'm not sure I'm happy about `pv.spherical_to_cartesian()` living in the main pyvista namespace. We have about 280 names there. I know this comes from `pv.features`, and there are already things like `pv.transform_vectors_sph_to_cart()` there that get pulled in. I haven't checked and I assume there's a star import `from pyvista.features`...

This is mostly musing and definitely unrelated to the PR. The only alternative would be to remove the star import there and keep `pv.features.spherical_to_cartesian()`, then again that would be hard to discover (if we _do_ want users to discover it). Of course others might want to keep it in the main namespace, that's also fine by me.

Reply:
I'm +1 to use this helper, but I tend to agree that we shouldn't add everything to the pyvista namespace.